CommandDotNet.Tests/BddTests/TestScenarios/GlobalOptionsInheritedScenarios.cs

```diff
@@ -0,0 +1,159 @@
+                new Given<Root>("help includes global options")
+                {
+                    WhenArgs = "-h",
+                    Then =
+                    {
+                        Help = @"Usage: dotnet testhost.dll [options] [command]
+
+Options:
+
+  -v | --version
+  Show version information
+
+  -h | --help
+  Show help information
+
+  --rootOpt         <TEXT>
+
+
+Commands:
+
+  Leaf
+
+Use ""dotnet testhost.dll [command] --help"" for more information about a command."
+                    }
+                },
+                new Given<Root>("help for sub-command includes local and inherited root global options")
+                {
+                    WhenArgs = "Leaf -h",
+                    Then =
+                    {
+                        Help = @"Usage: dotnet testhost.dll Leaf [options] [command]
+
+Options:
+
+  -h | --help
+  Show help information
+
+  --LeafOpt      <TEXT>
+
+  --rootOpt      <TEXT>
+
+
+Commands:
+
+  Do
+
+Use ""dotnet testhost.dll Leaf [command] --help"" for more information about a command."
+                    }
+                },
+                new Given<Root>("executing sub-command will parse and execute local global options")
+                {
+                    WhenArgs = "--rootOpt root Leaf --LeafOpt leaf Do --DoOpt a b",
+                    Then =
+                    {
+                        OutputsStrict = true,
+                        Outputs =
+                        {
+                            //new RootGlobalResult{RootOpt = "root"},
+                            new LeafGlobalResult{LeafOpt = "leaf"},
+                            new LeafDoResult{DoOpt = "a", DoArg = "b"}
+                        }
+                    }
+                },
+                new Given<Root>("global option can be specified after local command")
+                {
+                    WhenArgs = "Leaf Do --LeafOpt leaf --rootOpt root --DoOpt a b",
+                    Then =
+                    {
+                        OutputsStrict = true,
+                        Outputs =
+                        {
+                            //new RootGlobalResult{RootOpt = "root"},
+                            new LeafGlobalResult{LeafOpt = "leaf"},
+                            new LeafDoResult{DoOpt = "a", DoArg = "b"}
+                        }
+                    }
+                }
+            };
+
+        public class Root
+        {
+            private readonly string _rootOpt;
+
+            private TestOutputs _testOutputs;
+
+            [InjectProperty]
+            public TestOutputs TestOutputs
+            {
+                get => _testOutputs;
+                set
+                {
+                    _testOutputs = value;
+                    _testOutputs.Capture(new RootGlobalResult{RootOpt = _rootOpt});
+                }
+            }
+
+            public Root([Option(Inherited = true)] string rootOpt)
+            {
+                _rootOpt = rootOpt;
```

Comment:
@bilal-fazlani would you expect the root global options to be executed when nested sub-commands are executed?  is the behavior different for inherited vs not-inherited?

Looking through the docs, I think the current behavior is accurate but would like to confirm.